### DISCUSSION TITLE
hasValidKDocParamTags fails to distinguish non-KDoc tags (e.g. @type and @class), causing false negatives

### DISCUSSION THREAD

**Discussion by attila-fazekas** [LemonAppDev/konsist#1595] [Report a Problem] [answered]
### Tell us what version of Konsist you use.

0.17.0

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

[This](https://github.com/attila-fazekas/kolibrium/blob/a506104c97fb6f619f9bd4340e538aca70c60528/konsistTest/src/test/kotlin/dev/kolibrium/KDocTest.kt#L44) test uses Konsist's `hasValidKDocParamTags` function and fails with:

```
com.lemonappdev.konsist.core.exception.KoAssertionFailedException: Assert 'every public function with parameter(s) has param tag(s)' was violated (2 times). Invalid declarations:
├── Function xPath file:///Users/attila/projects/kolibrium/selenium/src/main/kotlin/dev/kolibrium/selenium/LocatorDelegates.kt:554:1
└── Function xPaths file:///Users/attila/projects/kolibrium/selenium/src/main/kotlin/dev/kolibrium/selenium/LocatorDelegates.kt:591:1
```

If you look at the example section in the referred [xPath](https://github.com/attila-fazekas/kolibrium/blob/a506104c97fb6f619f9bd4340e538aca70c60528/selenium/src/main/kotlin/dev/kolibrium/selenium/LocatorDelegates.kt#L561) and [xPaths](https://github.com/attila-fazekas/kolibrium/blob/a506104c97fb6f619f9bd4340e538aca70c60528/selenium/src/main/kotlin/dev/kolibrium/selenium/LocatorDelegates.kt#L598) functions KDoc, you will see `@type` and `@class` , which seem to make Konsist believe that these are KDoc tag declarations:
```
* Example usage:
 * ```
 * private val submitButton by xPath("//button[@type='submit']")
 * private val headerLink by xPath("//a[contains(@class, 'header-link')]", cacheLookup = false)
 * private val actionButton by xPath("//button[text()='Click Me']", readyWhen = { isDisplayed && isEnabled })
 * ```
```

If I remove the @ from both  `@type` and `@class` then the tests pass.

### Actual behaviour

`hasValidKDocParamTags` function fails to recognize that `@type` and `@class` are not KDoc tag declarations and causes my tests to fail.

### Expected behaviour

`hasValidKDocParamTags` function should only look for tags defined in [KDoc syntax](https://kotlinlang.org/docs/kotlin-doc.html#kdoc-syntax).

**Comment by nataliapeterwas** [comment, marked as answer]
Thank you for reporting the problem.
We will fix this in the next release: [PR](https://github.com/LemonAppDev/konsist/pull/1598) :)

**Comment by attila-fazekas** [comment]
Hi Natalia,
Awesome, thank you very much! 🙏

**Reply by igorwojda** [reply]
Fixed in [0.17.1](https://github.com/LemonAppDev/konsist/releases/tag/v0.17.1)
